### ISSUE TITLE
How to use this pretrained clinical BERT for multi class clinical text classification?

### ISSUE THREAD

**Issue by ghost** [kexinhuang12345/clinicalBERT#3]
Firstly, thanks for releasing the pre trained clinical bert models. I tried to use the pretrained clinical BERT model for multi class classification, but I got this error.

**RuntimeError:** Error(s) in loading state_dict for BertForSequenceClassification:
	**size mismatch for classifier.weight:** copying a param with shape torch.Size([2, 768]) from checkpoint, the shape in current model is torch.Size([181, 768]).
	**size mismatch for classifier.bias**: copying a param with shape torch.Size([2]) from checkpoint, the shape in current model is torch.Size([181]).

From the error message, what I understood is, the pretrained Clinical BERT model can be used  only for binary classification tasks.  

How to use this pre trained clinical BERT model for any multi class clinical text classification?

**Comment by kexinhuang12345** [comment]
Hi there,

Thank you for your interest in our work!

1. change "1" to "181", we were doing binary classification, so we just output one scalar and uses sigmoid function with BCE loss. 

https://github.com/kexinhuang12345/clinicalBERT/blob/e18eb857515e16887f896e897a52e7d666952d49/run_readmission.py#L513

2. you also need to change the "BertForSequenceClassification" class, to change sigmoid to softmax function and also change the corresponding loss to cross entropy loss.

Hope this helps!!

**Comment by ghost** [comment]
Thanks for the reply.  But, I ran my code after making the necessary changes for multi class classification. 

I downloaded ClinicalBERT from this repository https://github.com/EmilyAlsentzer/clinicalBERT
and did multi class text classification without any errors. In this case, when I loaded the model, only the model weights are copied but not the classifier weights. So, I can use this model for binary as well as multi class classification without making any changes.

But, when I tried with your ClinicalBERT embeddings, error is occuring because along with the model, the classifier weights are also copied. (Correct me if I'm wrong)

How can i load only the model weights excluding the classifier weights?

**Comment by kexinhuang12345** [comment]
Hi, I found this online:

pretrained_dict = torch.load(model_path.bin)
model_dict = model.state_dict() # your model

# 1. filter out unnecessary keys
pretrained_dict = {k: v for k, v in pretrained_dict.items() if k in model_dict}
# 2. overwrite entries in the existing state dict
model_dict.update(pretrained_dict) 
# 3. load the new state dict
model.load_state_dict(pretrained_dict)

Let me know if this works! So that I can update on the README to allow other people use it as well, thanks!

**Comment by ghost** [comment]
Thanks for the information. I will try.

**Comment by kexinhuang12345** [comment]
Close it for now, let me know if it does not work
